tutorials/llm/llama/domain-adaptive-pretraining/code/get_high_freq_tokens.py

```diff
@@ -0,0 +1,100 @@
+import json
+import sys
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def binary_search(arr, low, high, bar=0.98):
```

Comment:
Several qs about the use of binary search here:
1. the input to binary search is desc sorted frequencies, (e.g. [[4, 3, 3, 3, 3, 3, 3, 3, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]]), but low is the min freq(1) high being max freq(4). The binary search operates on indices. why is minfreq and max freq passed in as low and high? For the above freq list input, the output of this binary_search func is 3, which doesn't look correct to me according to "top-K tokens in a way that their cumulative frequency accounts for approximately 98%".
2. What about using `bisect` in python? Could we first calc the cumulative sum and then use `bisect` to perform binary search directly? The binary_search implementation is recursive and also calc `arr[0:mid].sum()` every recursive step, could be costly.

Reply:
@jvamaraju can you pls look into this and switch to bisect if possible?